### PR TITLE
resource/ssm_patch_baseline: Add compliance level to patch approval rules

### PR DESCRIPTION
**Pull request by pmorton**
Hi All - This pull request includes an update to support per patch rule compliance levels. 
- Documentation has been updated
- Acceptance tests have been updated and run on the SSM provider

Please let me know if you need anything else to get this merged in. 🍻

### CODE REVIEW

File: aws/resource_aws_ssm_patch_baseline.go [hunk 1]
```diff
@@ -57,6 +57,13 @@ func resourceAwsSsmPatchBaseline() *schema.Resource {
 							Required: true,
 						},
 
+						"compliance_level": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      "UNSPECIFIED",
```
[the remark this answers is not in the document]
Comment (pmorton) [reply aws/resource_aws_ssm_patch_baseline.go]: I have updated the code to move all of the levels and operating systems into constants. That should be contentious. I also added a constant-like variable to encapsulate the list of acceptable values. I am happy to revert this if needed....

File: aws/resource_aws_ssm_patch_baseline.go [hunk 2]
```diff
@@ -306,6 +313,7 @@ func expandAwsSsmPatchRuleGroup(d *schema.ResourceData) *ssm.PatchRuleGroup {
 		rule := &ssm.PatchRule{
 			ApproveAfterDays: aws.Int64(int64(rCfg["approve_after_days"].(int))),
 			PatchFilterGroup: filterGroup,
+			ComplianceLevel:  aws.String(rCfg["compliance_level"].(string)),
```
[the remark this answers is not in the document]
Comment (Ninir) [reply aws/resource_aws_ssm_patch_baseline.go]: You're totally right, missed the defaultFunc, nevermind :)

File: aws/resource_aws_ssm_patch_baseline.go [hunk 3]
```diff
@@ -10,6 +10,38 @@ import (
...
 )
 
+const (
+	ssmPatchComplianceLevelUnspecified   = "UNSPECIFIED"
+	ssmPatchComplianceLevelInformational = "INFORMATIONAL"
+	ssmPatchComplianceLevelLow           = "LOW"
+	ssmPatchComplianceLevelMedium        = "MEDIUM"
+	ssmPatchComplianceLevelHigh          = "HIGH"
+	ssmPatchComplianceLevelCritical      = "CRITICAL"
+)
```
Comment (Ninir) [inline aws/resource_aws_ssm_patch_baseline.go]: I was thinking of the existing ones, provided by the SDK: https://github.com/terraform-providers/terraform-provider-aws/blob/master/vendor/github.com/aws/aws-sdk-go/service/ssm/api.go#L27668-L27686 😄 

and for patch OS: https://github.com/terraform-providers/terraform-provider-aws/blob/master/vendor/github.com/aws/aws-sdk-go/service/ssm/api.go#L27615-L27627

Could you make the change? last thing remaining I think! :)

Comment (pmorton) [reply aws/resource_aws_ssm_patch_baseline.go]: @Ninir Done!

```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run=TestAccAWSSSMPatchBaseline* -timeout 120m
?   	github.com/terraform-providers/terraform-provider-aws	[no test files]
=== RUN   TestAccAWSSSMPatchBaseline_basic
--- PASS: TestAccAWSSSMPatchBaseline_basic (23.50s)
=== RUN   TestAccAWSSSMPatchBaselineWithOperatingSystem
--- PASS: TestAccAWSSSMPatchBaselineWithOperatingSystem (22.57s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	46.102s
```

**Pull request merged**